packages/horizon/contracts/staking/HorizonStaking.sol

```diff
@@ -286,7 +286,10 @@ contract HorizonStaking is HorizonStakingBase, IHorizonStakingMain {
 
         // Delegation pool must exist before adding tokens
         DelegationPoolInternal storage pool = _getDelegationPool(serviceProvider, verifier);
-        require(pool.shares > 0, HorizonStakingInvalidDelegationPool(serviceProvider, verifier));
+        require(
+            pool.shares > 0 || pool.sharesThawing > 0,
```

Comment:
Why are we changing this? If `pool.tokens == pool.tokensThawing` won't we be adding tokens making it so next delegate won't initialize the pool again and be calculated with `pool.shares = 0`?

Reply:
My intention was to allow recovering an invalid pool when the entire shares were thawing. So `pool.shares = 0` and `pool.sharesThawing != 0`. But I later changed the slashing so that it forces `pool.sharesThawing = 0` if the pool is slashed 100%, so this scenario would never happen. I'll remove this change

Reply:
good catch btw